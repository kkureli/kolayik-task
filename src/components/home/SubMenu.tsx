import React, {FC} from 'react';
import {Image, StyleSheet, TouchableOpacity, View} from 'react-native';
import {SubMenuModel} from '../../models';
import Typography from '../common/Typography';

interface SubMenuProps {
  menuList: SubMenuModel[];
}

interface RenderSubMenuProps {
  item: SubMenuModel;
}

const RenderSubMenu: FC<RenderSubMenuProps> = ({item}) => {
  return (
    <TouchableOpacity
      style={[styles.subContainer, styles.row]}
      onPress={() => item.onPress()}>
      <View style={styles.row}>
        <Image style={styles.icon} source={item.icon} />
        <Typography style={{marginLeft: 20}}>{item.title}</Typography>
      </View>
      <Image
        resizeMode="contain"
        style={styles.nextIcon}
        source={require('../../assets/icons/common/nextIcon.png')}
      />
    </TouchableOpacity>
  );
};

const SubMenu: FC<SubMenuProps> = ({menuList}) => {
  return (
    <View style={styles.container}>
      {menuList.map(item => (
        <RenderSubMenu key={item.id} item={item} />
      ))}
    </View>
  );
};

export default SubMenu;

const styles = StyleSheet.create({
  container: {
    marginTop: 15
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  icon: {
    width: 24,
    height: 24
  },
  nextIcon: {
    width: 20,
    height: 20
  },
  subContainer: {
    marginVertical: 15,
    marginHorizontal: 20
  }
});
