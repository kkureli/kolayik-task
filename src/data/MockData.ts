import {ContactMethods, SubMenuModel, User} from '../models';
const avatar = require('../assets/images/Avatar.png');

const mockUser: User = {
  image: avatar,
  name: 'Adnan Burhan',
  title: 'Product Designer',
  email: 'adnan@burhan.com',
  phoneNumber: 5360700195,
  department: 'Product',
  location: 'Istanbul',
  employmentStartDate: '25 Jan 2019',
  manager: 'Yunus Türkoğlu'
};

const contactOptions: ContactMethods[] = [
  {
    title: 'Email',
    icon: require('../assets/icons/contactOptions/emailIcon.png'),
    id: 'email'
  },
  {
    title: 'Call',
    icon: require('../assets/icons/contactOptions/phoneCallIcon.png'),
    id: 'call'
  }
];

const subMenu: SubMenuModel[] = [
  {
    title: 'Requests',
    icon: require('../assets/icons/subMenuIcons/requestsIcon.png'),
    id: '1',
    onPress: () => console.log('navigation.navigate(RequestsScreen)')
  },
  {
    title: 'Information',
    icon: require('../assets/icons/subMenuIcons/infoIcon.png'),
    id: '2',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Career',
    icon: require('../assets/icons/subMenuIcons/careerIcon.png'),
    id: '3',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Leaves',
    icon: require('../assets/icons/subMenuIcons/leavesIcon.png'),
    id: '4',
    onPress: () => console.log('navigation.navigate(LeavesScreen)')
  },
  {
    title: 'Assets',
    icon: require('../assets/icons/subMenuIcons/assetsIcon.png'),
    id: '5',
    onPress: () => console.log('navigation.navigate(AssetsScreen)')
  },
  {
    title: 'Documents',
    icon: require('../assets/icons/subMenuIcons/documentsIcon.png'),
    id: '6',
    onPress: () => console.log('navigation.navigate(DocumentsScreen)')
  }
];

export {mockUser, subMenu, contactOptions};
