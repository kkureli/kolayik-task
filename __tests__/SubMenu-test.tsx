import React from 'react';
import renderer from 'react-test-renderer';
import SubMenu from '../src/components/home/SubMenu';
import {subMenu} from '../src/data/MockData';

test('subMenu renders correctly', () => {
  const tree = renderer.create(<SubMenu menuList={subMenu} />).toJSON();
  expect(tree).toMatchSnapshot();
});
