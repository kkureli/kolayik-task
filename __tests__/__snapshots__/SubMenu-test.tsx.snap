// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subMenu renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/requestsIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Requests
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/infoIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Information
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/careerIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Career
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/leavesIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Leaves
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/assetsIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Assets
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/subMenuIcons/documentsIcon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        margin={null}
        padding={null}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginLeft": 20,
          }
        }
      >
        Documents
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/common/nextIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
